rust-runtime/aws-smithy-client/src/builder.rs

```diff
@@ -10,6 +10,14 @@ use aws_smithy_http::result::ConnectorError;
 use aws_smithy_types::timeout::TimeoutConfig;
 use std::sync::Arc;
 
+#[derive(Clone, Debug)]
+enum BuilderRetryConfig<R> {
+    // Since generic specialization isn't a thing yet, we have to keep a copy of the retry
+    // config around when using standard retry so that it can be properly validated.
+    Standard { has_retry: bool, policy: R },
+    Overridden(R),
```

Comment:
I think adding some extra info on what gets lost when overriding the retry config would be helpful to understanding how this all works/gets used

Reply:
I think the latest refactoring of the builder has improved this. Let me know if you think some additional comments should be added somewhere.

Reply:
I think it's good now. Your solution is both clever and explicit.